### PR TITLE
Implement global alignment constants

### PR DESCRIPTION
**Pull request by rboucher43**
[comment]: <> (Please tell us something about this pull request)

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

[comment]: <> ( What does this PR do? Linking to talk in software meeting encouraged )
Adds alignment parameters to Acts local to global transformations, and takes alignment parameters from the database. Alignment parameters are currently all zero in the database. 

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

### CODE REVIEW

**Comment by pinkenburg** [comment]
Please follow the include file conventions in https://wiki.sphenix.bnl.gov/index.php/Codingconventions#Include_Files
Breaking them can lead to very hard (means impossible for even most expert users) to debug issues
Please remove -L$(ROOTSYS)/lib from the LIBADD, it is already in LDADD (where the -L flags belong)

File: offline/packages/trackreco/PHActsTrkFitter.cc [hunk 1]
```diff
@@ -469,7 +479,88 @@ SourceLinkVec PHActsTrkFitter::getSourceLinks(TrackSeed* track,
...
+	  float globphi = atan2(global(1),global(0))*180.0/M_PI;
+	  std::cout << "Check in TrkFitter: global phi " << globphi << " hitsetkey: " << hitsetkey <<" global: " << std::endl << global << std::endl;
+
+	  auto x = cluster->getLocalX() * 10.0;   // mm
+	  auto y = cluster->getLocalY() * 10.0;
+	  
+	  if(trkrid == TrkrDefs::tpcId)
+	    {
+	      // must convert local Y from cluster average time of arival to local cluster z position
+	      double drift_velocity = 8.0e-3;  // cm/ns
```
Comment (osbornjd) [inline offline/packages/trackreco/PHActsTrkFitter.cc:502]: We should obtain this from the globally settable variable in ActsGeometry

**Review by osbornjd — CHANGES_REQUESTED** [review]
1. In general there is a lot of old code that is commented out - we should remove it completely to not confuse ourselves later (e.g. in ActsGeometry). 
2. A lot of the methods in ActsGeometry had their constness removed while trying to dig through an Acts compiler error about constness. But, if I recall, the compiler error didn't actually have to do with the constness in our class. If so, we should reimplement them. @adfrawley any comment on this, since I think (if I remember correctly) I talked to you about it?
3. We should also understand the drop in efficiency and pT resolution that Jenkins sees. It is probably related to the z position of the TPC clusters, as in the TPC QA the cluster z residuals look almost inverted.

Otherwise this looks like a great PR, I made a few other small comments but this is excellent to see!

**Comment by sPHENIX-GitHub-l** [comment]
Hi Joe,

I don't recall the part about constness not being the problem. If I recall correctly,  the problem we had was that we could not update the geometry context if the ActsTrackingGeometry class was const, since const classes cannot have non-const members. I had to remove the constness from all members to get things to compile. Possibly I misunderstood something.

You are correct about map2, that was overlooked when deleting some debugging code. I agree, I don't know how it compiled.
We will fix the ActsGeometry include.
Yes, sPHENIXActsDetectorElement::use_alignment is obsolete, we missed it.

The block of code in PHActsTrkFitter enabled by the test_transform flag was never intended to be left there permanently. But we will fix that, since we decided to keep that block there for a while.

Thanks for the useful comments!
Tony


________________________________

**Comment by osbornjd** [comment]
Okay, I couldn't remember on the const issue. That's fine then, I just needed a refresher. 

My guess is once the TPC z cluster residual issue gets understood it will fix the issues at the track level.

**Comment by adfrawley** [comment]
I do not see any reported issues that are related to this PR.

Tony
________________________________

**Comment by osbornjd** [comment]
Are the changes in the TPC cluster residuals understood? That (I suspect) negatively impacts the tracking and vertexing QA, especially in the z direction.

**Comment by osbornjd** [comment]
Looks good to me, @pinkenburg I assume the overlap failures are from the ongoing hcal issue so I will go ahead and merge this.

**Pull request merged**